frontend/src/component/common/util.ts

```diff
@@ -90,6 +91,40 @@ export function updateWeight(variants: IFeatureVariant[], totalWeight: number) {
     });
 }
 
+export function updateWeightEdit(
+    variants: IFeatureVariantEdit[],
+    totalWeight: number
+) {
+    if (variants.length === 0) {
+        return [];
+    }
+    const { remainingPercentage, variableVariantCount } = variants.reduce(
+        ({ remainingPercentage, variableVariantCount }, variant) => {
+            if (variant.weight && variant.weightType === weightTypes.FIX) {
+                remainingPercentage -= Number(variant.weight);
+            } else {
+                variableVariantCount += 1;
+            }
+            return {
+                remainingPercentage,
+                variableVariantCount,
+            };
+        },
+        { remainingPercentage: totalWeight, variableVariantCount: 0 }
+    );
+
+    const percentage = parseInt(
+        String(remainingPercentage / variableVariantCount)
+    );
+
+    return variants.map(variant => {
+        if (variant.weightType !== weightTypes.FIX) {
+            variant.weight = percentage;
+        }
+        return variant;
+    });
```

Comment:
This is probably (not sure), the same logic we have in the backend. Is there a way that can be shared? Just to avoid having it duplicated, because that means we need to keep them in sync